pkg/exporter/sumologicexporter/config.go

```diff
@@ -106,6 +109,25 @@ func CreateDefaultHTTPClientSettings() confighttp.HTTPClientSettings {
 
 func (cfg *Config) Validate() error {
 
+	switch cfg.CompressEncoding {
+	case GZIPCompression:
+	case NoCompression:
+	case DeflateCompression:
+
+	default:
+		return fmt.Errorf("invalid compression encoding type: %v", cfg.HTTPClientSettings.Compression)
+	}
+
+	switch cfg.HTTPClientSettings.Compression {
+	case GZIPCompression:
+	case NoCompression:
+	case DeflateCompression:
+	case ZSTDCompression:
```

Comment:
Since we know Zstd is currently bugged, we should prevent users from setting it until the upstream issue is resolved.

Reply:
A thought on this, could we allow the user to configure zstd with the new `Compress` config option and link to the known issue with memory usage? `CompressEncoding` wouldn't support this option

Reply:
We can just return an error. Why would that be a problem?

Reply:
The fix for the upstream issue may take a while, and I was wondering if we could include zstd support in the next release and mark it experimental, with a link to the bug. That way we may get some customer feedback as well while I work on the upstream issue. `gzip` will remain as the default and if a customer runs into memory issues with `zstd` they can always go back to the default (gzip).

Reply:
I guess if it's just for testing, we can include it. But we should print a warning if the user picks it.